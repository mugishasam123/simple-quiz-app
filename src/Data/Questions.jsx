const Questions = [
  {
    text: "What is the capital of America?",
    options: [
      { id: 0, text: "New York City", isCorrect: false },
      { id: 1, text: "Boston", isCorrect: false },
      { id: 2, text: "Santa Fe", isCorrect: false },
      { id: 3, text: "Washington DC", isCorrect: true },
    ],
  },
  {
    text: "What year was the Constitution of America written?",
    options: [
      { id: 0, text: "1787", isCorrect: true },
      { id: 1, text: "1776", isCorrect: false },
      { id: 2, text: "1774", isCorrect: false },
      { id: 3, text: "1826", isCorrect: false },
    ],
  },
  {
    text: "Who was the second president of the US?",
    options: [
      { id: 0, text: "John Adams", isCorrect: true },
      { id: 1, text: "Paul Revere", isCorrect: false },
      { id: 2, text: "Thomas Jefferson", isCorrect: false },
      { id: 3, text: "Benjamin Franklin", isCorrect: false },
    ],
  },
  {
    text: "What is the largest state in the US?",
    options: [
      { id: 0, text: "California", isCorrect: false },
      { id: 1, text: "Alaska", isCorrect: true },
      { id: 2, text: "Texas", isCorrect: false },
      { id: 3, text: "Montana", isCorrect: false },
    ],
  },
  {
    text: "Which of the following countries DO NOT border the US?",
    options: [
      { id: 0, text: "Canada", isCorrect: false },
      { id: 1, text: "Russia", isCorrect: true },
      { id: 2, text: "Cuba", isCorrect: true },
      { id: 3, text: "Mexico", isCorrect: false },
    ],
  },
];

export default Questions;
