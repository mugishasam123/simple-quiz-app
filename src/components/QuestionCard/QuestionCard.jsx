import React from "react";

const QuestionCard = ({ optionClicked, option }) => {
  return (
    <div
      className="mt-[8px] w-full md:w-[70%] text-center text-gray-500 bg-[#151A30] p-3 border border-gray-700 rounded-full text-[20px] hover:cursor-pointer hover:bg-[#281A39] hover:text-gray-200"
      key={option.id}
      onClick={() => optionClicked(option.isCorrect)}
    >
      {option.text}
    </div>
  );
};

export default QuestionCard;
