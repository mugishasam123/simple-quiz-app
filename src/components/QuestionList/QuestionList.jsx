import React, { useState } from "react";
import Questions from "../../Data/Questions";

const QuestionList = () => {
  

  return (
    <div >
   
    </div>
  );
};

export default QuestionList;
