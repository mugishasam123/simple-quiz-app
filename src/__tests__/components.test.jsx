import { describe, expect, test } from "vitest";
import { render, screen } from "@testing-library/react";
import QuestionCard from "../components/QuestionCard/QuestionCard.jsx";
import QuestionList from "../components/QuestionList/QuestionList.jsx";
import Questions from "../Data/Questions.jsx";

describe("Accordion test", () => {
  test("testing if QuestionCard is rendered", () => {
    const option = { id: 0, text: "New York City", isCorrect: false };
    const optionClicked = () => {
      console.log("this is questioncard");
    };

    expect(
      render(<QuestionCard option={option} optionClicked={optionClicked} />)
    ).toMatchSnapshot();
  });

  test("if QuestionList renders correctly", () => {
    const currentQuestion = 1;
    const optionClicked = () => {
      console.log("this is questionlist");
    };
    expect(
      render(
        <QuestionList
          optionClicked={optionClicked}
          Questions={Questions}
          currentQuestion={currentQuestion}
        />
      )
    ).toMatchSnapshot();
  });
});
