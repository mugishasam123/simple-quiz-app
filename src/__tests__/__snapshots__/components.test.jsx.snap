// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Questions components > if QuestionList renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col items-center gap-3 list-none "
      >
        <div
          class="mt-[8px] w-full md:w-[70%] text-center text-gray-500 bg-[#151A30] p-3 border border-gray-700 rounded-full text-[20px] hover:cursor-pointer hover:bg-[#281A39] hover:text-gray-200"
        >
          1787
        </div>
        <div
          class="mt-[8px] w-full md:w-[70%] text-center text-gray-500 bg-[#151A30] p-3 border border-gray-700 rounded-full text-[20px] hover:cursor-pointer hover:bg-[#281A39] hover:text-gray-200"
        >
          1776
        </div>
        <div
          class="mt-[8px] w-full md:w-[70%] text-center text-gray-500 bg-[#151A30] p-3 border border-gray-700 rounded-full text-[20px] hover:cursor-pointer hover:bg-[#281A39] hover:text-gray-200"
        >
          1774
        </div>
        <div
          class="mt-[8px] w-full md:w-[70%] text-center text-gray-500 bg-[#151A30] p-3 border border-gray-700 rounded-full text-[20px] hover:cursor-pointer hover:bg-[#281A39] hover:text-gray-200"
        >
          1826
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col items-center gap-3 list-none "
    >
      <div
        class="mt-[8px] w-full md:w-[70%] text-center text-gray-500 bg-[#151A30] p-3 border border-gray-700 rounded-full text-[20px] hover:cursor-pointer hover:bg-[#281A39] hover:text-gray-200"
      >
        1787
      </div>
      <div
        class="mt-[8px] w-full md:w-[70%] text-center text-gray-500 bg-[#151A30] p-3 border border-gray-700 rounded-full text-[20px] hover:cursor-pointer hover:bg-[#281A39] hover:text-gray-200"
      >
        1776
      </div>
      <div
        class="mt-[8px] w-full md:w-[70%] text-center text-gray-500 bg-[#151A30] p-3 border border-gray-700 rounded-full text-[20px] hover:cursor-pointer hover:bg-[#281A39] hover:text-gray-200"
      >
        1774
      </div>
      <div
        class="mt-[8px] w-full md:w-[70%] text-center text-gray-500 bg-[#151A30] p-3 border border-gray-700 rounded-full text-[20px] hover:cursor-pointer hover:bg-[#281A39] hover:text-gray-200"
      >
        1826
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Questions components > testing if QuestionCard is rendered 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mt-[8px] w-full md:w-[70%] text-center text-gray-500 bg-[#151A30] p-3 border border-gray-700 rounded-full text-[20px] hover:cursor-pointer hover:bg-[#281A39] hover:text-gray-200"
      >
        New York City
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mt-[8px] w-full md:w-[70%] text-center text-gray-500 bg-[#151A30] p-3 border border-gray-700 rounded-full text-[20px] hover:cursor-pointer hover:bg-[#281A39] hover:text-gray-200"
    >
      New York City
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
